### DISCUSSION TITLE
Feature request: Auto-detect ascii armored vs. binary

### DISCUSSION THREAD

**Discussion by chadmyers** [openpgpjs/openpgpjs#1786] [Ideas]
When decrypting a PGP file from a stream, I may not know whether it's ASCII-armored or binary.  It'd be nice if openpgpjs auto-detected it and did the right thing vs. me having to use `armoredMessage` or `binaryMessage`

**Comment by twiss** [comment]
Hi :wave: It's unlikely that we'll implement this in OpenPGP.js, as in most cases the application will know whether it's armored or not and if it doesn't know, it should be relatively simple to check. E.g. you could check whether it's an armored message by peeking into the file stream and checking whether it contains `-----BEGIN PGP MESSAGE-----`, or you could check whether it's a valid binary message by checking `(peekedBytes[0] & 0x80) !== 0`.
Alternatively, you could try first to parse it as binary and then if that fails, try to parse it as armored, e.g. like this: https://github.com/openpgpjs/sop-openpgpjs/blob/ad3a8ffd032f275d98160985c38bda212916d352/src/commands/decrypt.js#L9-L19.
